### PR TITLE
change CounterVec to use Counter in the Kubelet's Pod Lifecycle Event…

### PR DESCRIPTION
**Pull request by yiyang5055**
**What type of PR is this?**

 /kind cleanup

**What this PR does / why we need it**:

in the Kubelet's Pod Lifecycle Event Generator, we not use label name with CounterVec,  PLEGDiscardEvent  just use to tracks the numer of discarding events, so should use Counter.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
change CounterVec to Counter about  PLEGDiscardEvent
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by yiyang5055** [comment]
area/kubelet

File: pkg/kubelet/metrics/metrics.go [hunk 1]
```diff
@@ -165,17 +165,16 @@ var (
 			StabilityLevel: metrics.ALPHA,
 		},
 	)
-	// PLEGDiscardEvents is a Histogram that tracks the duration (in seconds) it takes for discarding events in the Kubelet's
-	// Pod Lifecycle Event Generator (PLEG).
-	PLEGDiscardEvents = metrics.NewCounterVec(
+	// PLEGDiscardEvents is a Counter that tracks the number of discarding events in the Kubelet's Pod Lifecycle Event Generator (PLEG).
+	PLEGDiscardEvents = metrics.NewCounter(
```
Comment (mattjmcnaughton) [inline pkg/kubelet/metrics/metrics.go]: Do we need to go through some form of deprecation process to change the type of this metric? I.e. would this change not cause breakages for anyone who currently consumes `PLEGDiscardEvents`?

Does the `StabilityLevel: metrics.ALPHA` designation have any impact on my question above?

**Review by mattjmcnaughton — COMMENTED** [review]
/ok-to-test

Thanks for your pr!

Curious to hear from others on whether this change is desirable - if we do decide its desirable, I have a question on whether we have concerns about it breaking existing use cases.

File: pkg/kubelet/metrics/metrics.go (hunk 1 above)
Comment (yiyang5055) [reply pkg/kubelet/metrics/metrics.go]: ALPHA metrics have no stability guarantees，and it just use in PLEG relist() function, so no impact for other one

Comment (mattjmcnaughton) [reply pkg/kubelet/metrics/metrics.go]: :+1:

**Review by mattjmcnaughton — COMMENTED** [review]
/assign @dashpole 

I defer to you whether this change is beneficial :)

**Comment by dashpole** [comment]
yiyang5055@ Does this change the output of scraping this endpoint?  It seems like a Vec type metric without any dimensions should be equivalent to a non-vec metric

**Comment by yiyang5055** [comment]
> yiyang5055@ Does this change the output of scraping this endpoint? It seems like a Vec type metric without any dimensions should be equivalent to a non-vec metric

@dashpole  No,  the output of scraping this endpoint is the same as change before. the vec type metric whitout labelNames parameter when NewCounterVec() output format is the same as no-vec type metric.
PLEGDiscardEvents is vec type metric and output:
```
# HELP kubelet_pleg_discard_events The number of discard events in PLEG.
# TYPE kubelet_pleg_discard_events counter
kubelet_pleg_discard_events 24
```
then vec type metric output:
```
# HELP kubelet_pleg_discard_events The number of discard events in PLEG.
# TYPE kubelet_pleg_discard_events counter
kubelet_pleg_discard_events 24
```

**Comment by dashpole** [comment]
given that this doesn't actually change the metric at all, and simplifies our code (ever so slightly), I think this is good to merge.
/lgtm
/approve

**Comment by yiyang5055** [comment]
pinging @derekwaynecarr

**Comment by derekwaynecarr** [comment]
/approve

Thanks!

**Pull request merged**